src/core/domain/pedido.ts

```diff
@@ -19,6 +19,7 @@ export interface Pedido {
 export const statusDoPedido = {
   RASCUNHO: "Rascunho",
   AGUARDANDO_PAGAMENTO: "Aguardando pagamento",
+  FALHA_PAGAMENTO: "Falha no pagamento",
```

Comment:
Acredito que Falha de Pagamento deveria ser só um status de fatura. Até pelo fato de que se for reprocessar o pagamento seria somente relativo a fatura. O pedido segue aguardando o pagamento, porém a fatura teve uma falha de pagamento.

Faz sentido?

Reply:
Acho que se o pagamento teve problema faz sentido ter um status de erro, acho que faz mais sentido algo como "Pedico cancelado" e deixar esse apenas na fatura mesmo

Reply:
Mas e se o cliente quiser só reprocessar o pagamento? 

A gente já teria cancelado o pedido, não?

Reply:
faz sentido manter `aguardando o pagamento` no caso de reprocessamento do pagamento? o pedido só teria os estados de `aguardando pagamento` ou `pago`. e também um status/endpoint de `cancelado`?

Reply:
A gente tem esses status:

  RASCUNHO: "Rascunho",
  AGUARDANDO_PAGAMENTO: "Aguardando pagamento",
  AGUARDANDO_PREPARO: "Aguardando preparo",
  EM_PREPARO: "Em preparo",
  PRONTO: "Pronto",

Reply:
sim, manteríamos esses e adicionamos um de `cancelado` apenas para pedidos cancelados mesmo. o status `aguardando pagamento` só mudaria para `aguardando preparo` (sucesso no pagamento) ou `cancelado` (cancelamento do pedido). enquanto o pagamento estivesse sendo processado ou reprocessado o status do pedido se manteria como `aguardando pagamento` pois o status do pagamento e o status do pedido são processos diferentes. acham que faz sentido?

como por enquanto é fake checkout não precisamos pensar no status do pagamento (processamento, quantas tentativas, etc) nesse momento.